lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -334,3 +378,255 @@ def _delete_node(node_ctx: NodeContext, wait_delete: bool) -> None:
                 for node in environment.nodes.list()
             ]
         )
+
+
+class HyperVDevicePoolImpl(BaseDevicePoolImpl):
+    def __init__(
+        self,
+        node: RemoteNode,
+        assignable_device_script: PurePath,
+        runbook: HypervPlatformSchema,
+    ) -> None:
+        # Device Passthrough configs
+        # Mapping of Host Device Passthrough
+        self.AVAILABLE_HOST_DEVICES: Dict[
+            HostDevicePoolType, List[DeviceAddressSchema]
+        ] = {}
+        self.SUPPORTED_HOST_DEVICE_POOLTYPE = [
+            HostDevicePoolType.PCI_NIC,
+            HostDevicePoolType.PCI_GPU,
+        ]
+        self.POOL_TYPE_TO_DEVICE_PROPERTY = {
+            HostDevicePoolType.PCI_NIC: DeviceAddressSchema,
+            HostDevicePoolType.PCI_GPU: DeviceAddressSchema,
+        }
+        self._server = node
+        self._assignable_devices_script = assignable_device_script
+        self._hyperv_runbook = runbook
+
+    def _create_device_pool(
+        self,
+        pool_type: HostDevicePoolType,
+        vendor_id: str,
+        device_id: str,
+    ) -> None:
+        self.AVAILABLE_HOST_DEVICES[pool_type] = []
+        powershell = self._server.tools[PowerShell]
+        cmdlet = (
+            f"{self._assignable_devices_script} "
+            f"-vendorId {vendor_id} -deviceId {device_id}"
+        )
+        stdout = powershell.run_cmdlet(
+            cmdlet=cmdlet,
+            sudo=True,
+            force_run=True,
+        ).strip()
+
+        if stdout.find("The list is empty") >= 0:
```

Comment:
Move the whole logic into Python, it doesn't need the second time conversion.

Reply:
Sure will do it